R/parse.R

```diff
@@ -51,18 +51,35 @@ parse_delim <- function(x, delim, quote = "\"", ...) {
   )
 }
 
-parse_name_equals_value <- function(x) {
-  loc <- regexpr("=", x, fixed = TRUE)
-  pieces <- regmatches(x, loc, invert = TRUE)
-
+parse_name_equals_value <- function(x, safe = "&", quote = "\"") {
```

Comment:
I think I've figured out a faster way to do this which is also a little clearer (if more verbose): https://gist.github.com/hadley/2fd32cdc053099d62d56f5f28898ec95

The main insight I had is that main culprit for the slow speed in my approach is `regmatches()`, and for this simple case, it's easy enough to do by hand. I also named the function `parse_in_half()` as there's an existing function with that name that only works for scalars.

Reply:
Plus it looks like you get around the R magic from the `do.call(rbind, nested_list)`.